app/controllers/report_controller/reports/editor.rb

```diff
@@ -743,11 +726,11 @@ def gfv_pivots
 
   def gfv_sort
     @edit[:new][:order] = params[:sort_order] if params[:sort_order]
-    if params[:sort_group]                                    # If grouping changed,
+    if params[:sort_group] # If grouping changed,
       @edit[:new][:group] = params[:sort_group]
-      @refresh_div = "sort_div"                               # Resend the sort tab
+      @refresh_div = "sort_div" # Resend the sort tab
       @refresh_partial = "form_sort"
-      if @edit[:new][:chart_mode] = 'values' && !chart_mode_values_allowed?
+      if @edit[:new][:chart_mode] == 'values' && !chart_mode_values_allowed?
```

Comment:
This looks like you found a bug. Shall this go to Hammer?

Reply:
It seemed rather odd to me to have:
```ruby
if @edit[:new][:chart_mode] = 'values' && !chart_mode_values_allowed?
  @edit[:new][:chart_mode] = 'counts'
end
```
i.e. first assign `@edit[:new][:chart_mode] = 'values'` and right after that assign `@edit[:new][:chart_mode] = 'counts'` so I changed that.

Whether or not to have this in hammer -- right now I'd rather leave it as it is, should a bz in hammer show, we'll backport it then.